assets/sass/02-tools/mixins.scss

```diff
@@ -77,3 +77,21 @@
 		color: var(--button--color-text-active);
 	}
 }
+
+@mixin innerblock-margin-clear($container) {
+
+	// Clear the top margin for the first-child.
+	> #{$container} > *:first-child {
+		margin-top: 0;
+	}
+
+	// Last child that is not the appender.
+	> #{$container} > *:last-child:not(.block-list-appender) {
+		margin-bottom: 0;
+	}
+
+	// When selected, the last item becomes the second last because of the appender.
+	&.has-child-selected > #{$container} > *:nth-last-child(2) {
```

Comment:
```suggestion
	&.has-child-selected > #{$container} > *:nth-last-child(2),
	&.is-selected > #{$container} > *:nth-last-child(2) {
```

This fixes the selected parent issue I recorded— since selecting the container also adds the appender, we still need to clear this margin. Fixing this also seems to cut down on the has-child-selected delay.